[JavaScript] Improve Concept Exercise: strings (#1381)
exercism/v3: languages/javascript/exercises/concept/strings/.docs/introduction.md

Strings are useful for holding data that can be represented in text form.
There are two ways to access an individual character in a string.

> The first is the charAt() method:

Reviewer: ```suggestion
The first is the `charAt()` method:
```

```javascript
'cat'.charAt(1)
// => "a"
```

> The other way is to treat a string as an array-like object, where individual characters correspond to a numerical index(starts at zero):

Reviewer: ```suggestion
The other way is to treat a `string` as a list of characters, where individual characters correspond to a numerical index (starts at zero):
```

Reviewer: I''m okay lying to the student here. It's not individual characters, but code points.
